tests/test_device_aws.py

```diff
@@ -0,0 +1,329 @@
+"""Tests for DeviceAws.
+
+Here is one way to run it:
+
+First install the package in developer mode
+
+    $ pip install -e .
+
+Then use pytest to drive the tests
+
+    $ pytest tests
+"""
+
+from typing import Any
+
+from importlib import resources
+from unittest import mock
+from unittest import IsolatedAsyncioTestCase
+
+import pytest
+
+from blueair_api.device_aws import DeviceAws
+from blueair_api.model_enum import ModelEnum
+from blueair_api import http_aws_blueair
+from blueair_api import ir_aws as ir
+
+class FakeDeviceInfoHelper:
+    """Fake for the 'device info' interface of HttpAwsBlueAir class."""
+    def __init__(self, info: dict[str, Any]):
+        self.info = info
+
+    async def device_info(self, *args, **kwargs):
+        return self.info
+
+    async def set_device_info(self, device_uuid, service_name, action_verb, action_value):
+        # this function seems to be only updating the states consider rename the method.
+        # action_verb seems to be a type annotation:
+        # c.f. senml: https://www.rfc-editor.org/rfc/rfc8428.html#section-5
+        # the senml parsing library (utils.py) could use some additional love.
+        # to make it more conformal to the RFC standard.
+        for state in self.info['states']:
+            if state['n'] == service_name:
+                break
+        else:
+            state = {'n': service_name}
+            self.info['states'].append(state)
+        # Watch out: mutate after append produces desired mutation to info.
+        state[action_verb] = action_value
+
+class DeviceAwsTestBase(IsolatedAsyncioTestCase):
+
+    def setUp(self):
+
+        patcher = mock.patch('blueair_api.http_aws_blueair.HttpAwsBlueair', autospec=True)
+        self.api_class = patcher.start()
+        self.addCleanup(patcher.stop)
+        self.api = self.api_class(username="fake-username", password="fake-password")
+
+        self.device = DeviceAws(self.api, name_api="fake-name-api", uuid="fake-uuid", name="fake-name",
+mac="fake-mac", type_name='fake-type-name')
+
+        self.device_info_helper = FakeDeviceInfoHelper(
+           {"configuration": {"di" : {}, "ds" : {}, "dc" : {}, "da" : {},},
+            "sensordata": [],
+            "states": [],
+           })
+
+        self.api.device_info.side_effect = self.device_info_helper.device_info
+        self.api.set_device_info.side_effect = self.device_info_helper.set_device_info
+
+
+class DeviceAwsSetterTest(DeviceAwsTestBase):
+    """Tests for all of the setters."""
+
+    def setUp(self):
+        super().setUp()
+        fake = {"n": "n", "v": 0}
+        ir.query_json(self.device_info_helper.info, "configuration.dc").update({
+          "brightness": fake,
+          "fanspeed": fake,
+          "standby": fake,
+          "automode": fake,
+          "autorh": fake,
+          "childlock": fake,
+          "nightmode": fake,
+          "wickdrys": fake,
+        })
+
+    async def test_brightness(self):
```

Comment:
these test bodies are almost identical, but between the current form and pytest.mark.parameterize, I picked the more concrete style.

Reply:
I can't tell the difference, but like concrete in general for tests